### DISCUSSION TITLE
making "Button" class so sendButtons accept it and not plain object

### DISCUSSION THREAD

**Discussion by MoonLGH** [open-wa/wa-automate-nodejs#2706] [General]
[SendButtons](https://docs.openwa.dev/classes/api_Client.Client.html#sendButtons)
feels so off when it says it accept Button[] but instead it accept an object

**Comment by MoonLGH** [comment]
[Button Interface](https://github.com/open-wa/wa-automate-nodejs/blob/master/src/api/model/button.ts)

what about making buttons like something like a discord.js Embed class, so its acts like a builder

**Comment by smashah** [comment]
@MoonLGH I don't want to have to force users to use a whole class structure for an object without any methods.

There are a lot of standards that are people are used to and I want to make open-wa accessible to the most types of devs out there.

You can implement this on your own though:

```js
interface Button {
    id: string,
    text: string
}

class WButton implements Button{
id: string;
text: string;

constructor(id : string, text : string) {
    this.id = id;
    this.text = text
}
}

console.log(JSON.stringify(new WButton("id_1","Press")))
```

Might look into this for v5 if there's a large enough community demand for it.

**Comment by MoonLGH** [comment]
how about there is a builder class and it accepts both Button and object

**Comment by MoonLGH** [comment]
thinking about 
```ts
class ButtonBuilder implements Button {
  id: string;
  text: string;

  constructor(id: string, text: string) {
    this.id = id;
    this.text = text;
  }
  setText(text: string) {
    this.text = text;
  }
  setId(id: string) {
    this.id = id;
  }

  toJson() {
    if (!this.text || !this.id){
      throw 'please put a valid text and id of the button';
    }
    return { text: this.text, id: this.id };
  }
}

```
then 
```ts
public async sendButtons(to: ChatId, body : string | LocationButtonBody, buttons : ButtonBuilder[]|Button[], title ?: string, footer ?: string) : Promise<boolean | MessageId> {
    if(buttons[0] instancesOf WButton){
    buttons = buttons.map((b) => b.toJson());
    }
    return await this.pup(
      ({ to,  body, buttons, title, footer }) => {
        return WAPI.sendButtons(to, body, buttons, title, footer);
      },
      { to, body, buttons, title, footer }
    ) as Promise<boolean | MessageId>;
  }
  ```

**Comment by MoonLGH** [comment]
that way it accept both

**Comment by smashah** [comment]
@MoonLGH bring this up in the buttons channel so if people are interested in using a builder pattern. As long as the builder class implements the button interface then it should work without the change to `sendButtons`. I'm moving this to a discussion.

**Comment by MoonLGH** [comment]
just made #2707 pr, tell me something if something wrong
